### DISCUSSION TITLE
Concern re: future breaking change for modbus integration (#46591)

### DISCUSSION THREAD

**Discussion by bradkeifer** [home-assistant/architecture#560] [General] [answered]
I am concerned regarding the impact of a proposed future breaking change for the modbus integration as described in issue [#46591](https://github.com/home-assistant/core/pull/46591).
If I understand this proposed breaking change correctly, it is requiring the modbus connection details to exist in the same .yaml configuration file as the definitions for each and every entity associated with that connection. I make heavy use of packages to define my configuration, and it allows me to keep the modbus physical connectivity definitions separated from the definitions for the entities. I am concerned that the breaking change will constrain the utility of the system and significantly reduce the ability to make use of the packages capability that exists in home assistant configuration.

My request is that the modbus integration continues to support both the "old" and the "new" configuration styles as documented in issue [#46591](https://github.com/home-assistant/core/pull/46591).

**Comment by frenck** [comment, marked as answer]
This is not a feature request section.

Decisions on configurations and their styles/methods are documented in ADR-0007 and ADR-0010.

All integrations are going to follow those ADRs over time.

**Reply by bradkeifer** [reply]
OK, thanks @frenck
